src/scripting.md

```diff
@@ -5,66 +5,179 @@ eleventyNavigation:
+execAndWait document.querySelector('.chat-dialog > button').click()
 ```
 
-you could identify it as **id=lgnId1** , **name=loginId** or **tabindex=1**
+See more [commands](#recommended-commands) and [examples](#sample-scripts) below.
+
+## Recommended Commands
+
+For historical reasons, if a script step fails, it does so silently. If something doesn’t work, unfortunately, you won’t get any feedback why. If you’re new to scripting, we recommend using the following commands only. It’s easier to debug them as you can run them manually yourself.
+
+#### `navigate`
+Navigates the browser to the provided URL and waits for it to complete.
+Browser Support: IE, Chrome, Firefox, Safari
+
+```markup
+usage:   navigate <url>
+example: navigate http://webmail.aol.com
+
+<url> - URL to provide the browser for navigation (same as you would enter into the address bar)
+```
+
+#### `exec`
+Executes JavaScript.
+Browser Support: IE, Chrome, Firefox
+```markup
+usage:   exec <JavaScript code>
+example: exec window.setInterval('window.scrollBy(0,600)', 1000);
+```
+
+#### `execAndWait`
+Executes JavaScript and waits for the browser to complete any activity generated from the action.
+Browser Support: IE, Chrome, Firefox
+```markup
+usage:   execAndWait <JavaScript code>
+example: execAndWait window.setInterval('window.scrollBy(0,600)', 1000);
+```
+
+## Full Reference
+
+### Comments
+
+A line that starts with  `// ` is ignored. Use this for comments.
+
+```markup
+// Example comment
+navigate not-a-comment.com
+```
+
+### Selectors
+
+Commands like [`click`](#click), [`setValue`](#setvalue), and others operate on DOM elements. To select a DOM element, use the following selectors:
+
+#### `<attribute>=<value>`
+
+This selector selects elements based on their HTML attributes. For the following input:
 
-For form fields it is usually best to use the name attribute when it is available since that is what will be uploaded to the server but any attribute is fair game. The class attribute is special and is referenced as className instead of class. In addition to the DOM element attribute matching, there are two special attributes that you can use to match on the contents. innerText and innerHtml both of which will match the contents of the DOM element instead of it's attributes.
-For example:
 ```html
-<div dojoattachpoint="containerNode" class="label">Delete</div>
+<input type="email" id="emailField" class="input input-email">
 ```
-Can be identified by innerText=Delete. The matching is case sensitive and matches the full string.
 
-In order to suppress intermediate steps you need to make sure that data logging is disabled for the steps up to the ones you want to record. For example:
+the following selectors will work:
+
+* ✅ `type=email`
+* ✅ `id=emailField`
+
+and the following selectors will not:
+
+* ❌ `class=input input-email` (spaces inside attribute values are not supported)
+* ❌ `class="input input-email"` (not even if quoted)
```

Comment:
Is this correct? I can’t manage to make my `setValue` script work so I can’t test this.

Reply:
It is!